apps/condo/domains/miniapp/utils/b2bAppServiceUserAccess/server.utils.js

```diff
@@ -0,0 +1,217 @@
+        organization: { id: organizationId, deletedAt: null },
+        app: {
+            accessRights_some: {
+                accessRightSet: {
+                    [`canManage${pluralize.plural(listKey)}`]: true,
+                    deletedAt: null,
+                },
+                user: { id: user.id, type: 'service', deletedAt: null },
+                deletedAt: null,
+            },
+            deletedAt: null,
+        },
+        status: 'Finished',
+        deletedAt: null,
+    })
+
+    return !isEmpty(B2BAppContexts)
+}
+
+const isServiceUser = ({ authentication: { item: user } }) => {
+    return get(user, 'type') === SERVICE
+}
+
+const getRefSchemaName = (schemaConfig, listKey) => {
+    const pathToOrganizationId = get(schemaConfig, 'pathToOrganizationId', ['organization', 'id'])
+
+    if (!isArray(pathToOrganizationId) || isEmpty(pathToOrganizationId)) {
+        throw new Error('"pathToOrganizationId" must be not empty array!')
+    }
+    for (const pathPart of pathToOrganizationId) {
+        if (!isString(pathPart) || pathPart.trim().length < 1) {
+            throw new Error(`"pathToOrganizationId" must contain array of string! But was: ${pathToOrganizationId}`)
+        }
+    }
+
+    const schema = getSchemaCtx(listKey)
+
+    const schemaFields = get(schema, 'list._fields', {})
+    return get(schemaFields, [pathToOrganizationId[0], 'ref'], null)
+}
+
+/**
+ * Checks that service user can read objects of organization that is connected to linked B2B app
+ *
+ * @param args
+ * @return {Promise<Record<string, any>|false>}
+ */
+const b2bAppServiceUserCanReadObjects = async (args) => {
+    const { listKey } = args
+    if (!isServiceUser(args)) return false
+    const schemaConfig = get(B2B_APP_SERVICE_USER_ACCESS_AVAILABLE_SCHEMAS, listKey)
+    if (!isObject(schemaConfig)) return false
+    const canBeRead = get(schemaConfig, 'canBeRead', true)
+    if (!canBeRead) return false
+    return await canReadByServiceUser(args, schemaConfig)
+}
+
+/**
+ * Checks that service user can manage objects of organization that is connected to linked B2B app
+ *
+ * @param args
+ * @return {Promise<boolean>}
+ */
+const b2bAppServiceUserCanManageObjects = async (args) => {
+    const { listKey } = args
+    if (!isServiceUser(args)) return false
+    const schemaConfig = get(B2B_APP_SERVICE_USER_ACCESS_AVAILABLE_SCHEMAS, listKey)
+    if (!isObject(schemaConfig)) return false
+    const canBeManage = get(schemaConfig, 'canBeManage', true)
+    if (!canBeManage) return false
+    const refSchemaName = getRefSchemaName(schemaConfig, listKey)
+    return await canManageByServiceUser(args, schemaConfig, refSchemaName)
+}
+
+/**
+ * Merge (if it needed) two access filters or return false
+ *
+ * @param a {any}
+ * @param b {any}
+ * @return {Record<string, any>|{OR: [{AND: Record<string, any>[]},{AND: Record<string, any>[]}]}|boolean}
+ */
+const mergeAccessFilters = (a, b) => {
```

Comment:
It’s not a good idea to use some math to calculate accesses. It should be as match simple as possible. No any transformations and code logic

Reply:
removed this function